task/aws/resources/data_source_bucket.go

```diff
@@ -3,69 +3,60 @@ package resources
 import (
 	"context"
 	"fmt"
-	"path"
-
-	"terraform-provider-iterative/task/common"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"terraform-provider-iterative/task/common"
+	"terraform-provider-iterative/task/common/machine"
 )
 
 // NewExistingS3Bucket returns a new data source refering to a pre-allocated
 // S3 bucket.
-func NewExistingS3Bucket(client S3Client, credentials aws.Credentials, id string, region string, path string) *ExistingS3Bucket {
+func NewExistingS3Bucket(credentials aws.Credentials, storageParams common.RemoteStorage) *ExistingS3Bucket {
```

Comment:
Passing `common.RemoteStorage` to create a reference to an existing s3 bucket, instead of individual params.